library/spark/spark-accumulo-library/src/main/java/uk/gov/gchq/gaffer/sparkaccumulo/operation/handler/scalardd/ImportKeyValuePairRDDToAccumuloHandler.java

```diff
@@ -0,0 +1,86 @@
+import uk.gov.gchq.gaffer.operation.Operation;
+import uk.gov.gchq.gaffer.operation.OperationException;
+import uk.gov.gchq.gaffer.sparkaccumulo.operation.scalardd.ImportKeyValuePairRDDToAccumulo;
+import uk.gov.gchq.gaffer.sparkaccumulo.operation.utils.AccumuloKeyRangePartitioner;
+import uk.gov.gchq.gaffer.store.Context;
+import uk.gov.gchq.gaffer.store.Store;
+import uk.gov.gchq.gaffer.store.operation.handler.OperationHandler;
+
+import java.io.ByteArrayInputStream;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.util.Comparator;
+
+public class ImportKeyValuePairRDDToAccumuloHandler implements OperationHandler<ImportKeyValuePairRDDToAccumulo, Void> {
+
+    private static final ClassTag<Key> KEY_CLASS_TAG = scala.reflect.ClassTag$.MODULE$.apply(Key.class);
+    private static final ClassTag<Value> VALUE_CLASS_TAG = scala.reflect.ClassTag$.MODULE$.apply(Value.class);
+    private static final Ordering<Key> ORDERING_CLASS_TAG = Ordering$.MODULE$.comparatorToOrdering(Comparator.<Key>naturalOrder());
+
+    @Override
+    public Void doOperation(final ImportKeyValuePairRDDToAccumulo operation, final Context context, final Store store) throws OperationException {
+        String outputPath = operation.getOutputPath();
+        if (null == outputPath || outputPath.isEmpty()) {
+            throw new OperationException("Option outputPath must be set for this option to be run against the accumulostore");
+        }
+        String failurePath = operation.getFailurePath();
+        if (null == failurePath || failurePath.isEmpty()) {
+            throw new OperationException("Option failurePath must be set for this option to be run against the accumulostore");
+        }
+        doOperation(operation, context, (AccumuloStore) store);
+        return null;
+    }
+
+    private void doOperation(final ImportKeyValuePairRDDToAccumulo operation, final Context context, final AccumuloStore store) throws OperationException {
+        AccumuloKeyRangePartitioner partitioner = new AccumuloKeyRangePartitioner(store);
+        OrderedRDDFunctions orderedRDDFunctions = new OrderedRDDFunctions(operation.getInput(), ORDERING_CLASS_TAG, KEY_CLASS_TAG, VALUE_CLASS_TAG, scala.reflect.ClassTag$.MODULE$.apply(Tuple2.class));
+        PairRDDFunctions pairRDDFunctions = new PairRDDFunctions(orderedRDDFunctions.repartitionAndSortWithinPartitions(partitioner), KEY_CLASS_TAG, VALUE_CLASS_TAG, ORDERING_CLASS_TAG);
+        pairRDDFunctions.saveAsNewAPIHadoopFile(operation.getOutputPath(), Key.class, Value.class, AccumuloFileOutputFormat.class, getConfiguration(operation));
+        ImportAccumuloKeyValueFiles importAccumuloKeyValueFiles = new ImportAccumuloKeyValueFiles.Builder().inputPath(operation.getOutputPath()).failurePath(operation.getFailurePath()).build();
+        store.execute(importAccumuloKeyValueFiles, context.getUser());
+    }
+
+    protected Configuration getConfiguration(final Operation operation) throws OperationException {
```

Comment:
Seems to be a repeat of code from ImportKeyValueJavaPairRDDToAccumuloHandler ? Can it be moved to a utils class?